conans/util/files.py

```diff
@@ -112,20 +112,28 @@ def save_append(path, content):
         handle.write(to_file_bytes(content))
 
 
-def save(path, content):
+def save(path, content, only_if_modified=False):
```

Comment:
As this is only used internally (`tools.save()` is not affected by this) and mostly for testing (I think). Would it be possible to benefit from a defaulted `only_if_modified=True`?